lib/pbench/cli/server/__init__.py

```diff
@@ -19,3 +23,12 @@ class CliContext:
 
 
 pass_cli_context = click.make_pass_decorator(CliContext, ensure=True)
+
+
+def config_setup(context: object, name: str) -> None:
+    config = PbenchServerConfig(context.config)
+    logger = get_pbench_logger(name, config)
+
+    # We're going to need the Postgres DB to track dataset state, so setup
+    # DB access.
+    Database.init_db(config, logger)
```

Comment:
Right now, it's the right call; soon it'll be wrong. It's that timey-wimey stuff...

Reply:
Turns out this line was not the problem, it was about the way I was calling it. I was calling the init_db separately for each command (create, update, delete etc) instead of under our broad click group `user_group` and thats why unit tests were not able to share the in memory db instance.

Reply:
Understood.  However, this one looks like a prime case where the merge would fail to pick up on the need to update it, so I figured I should call it out.